backend/sites/src/app/services/snapshot/snapshot.service.ts

```diff
@@ -72,6 +72,19 @@ export class SnapshotsService {
     }
   }
 
+  async getMostRecentSnapshot(siteId: string, userId: string) {
```

Comment:
We already have a method (getSnapshotsBySiteId) to return the snapshots for siteId and userId in desc order. IMO you can use that and fetch the first record.

Reply:
I agree that this is only a minor optimization, but I think it's useful to have around since it will be a common use case to fetch the most recent snapshot for the user/site. It's more ergonomic to call `snapshot = snapshotService.getMostRecentSnapshot(siteId, userId)` than it is to call `snapshots = snapshotService.getSnapshotsBySiteId(siteId, userId); snapshot = snapshots ? snapshot[0] : null`.